### ISSUE TITLE
Make CustomVariable.ini Tool

### ISSUE THREAD

**Issue by OrionSR** [sannybuilder/dev#4]
Consider adding a tool to Make CustomVariables.ini based on how the custom variables will be compiled. This is a difficult process to manage manually.

An application is to allow Sanny to manage the variables in a custom main.scm and produce a list that can be used to decompile that script with the same custom variables.

Another application is to update the current variables in a standard main,scm using the Alloc function.

**Comment by x87** [comment]
So given a decompiled source file you want to be able to get a list of unnamed global variables and replace them with a custom names / produce a new CustomVariables.ini for this file?

**Comment by OrionSR** [comment]
Not quite. Unnamed variables are another issue but there might be an application here anyway.

The primary goal is to produce a list of all named variables in the format of CustomVariables.ini, so a custom script can be decompiled using the same variable names. This list would (optionally?) include:
 * Undefined custom variables used
 * Defined variables used
 * Defined variables unused
Options for unnamed variables used and/or unused would certainly be handy.

Given a quick way to generate new CustomVariables files, it occurred to me that an easy way to name variables in a standard main would be to use the Alloc function in-line as I work out the purpose of the variable, in a proper variable table in a header, or in an include file, and that these would count as named variables.

**Comment by x87** [comment]
If you enable "**Add extra info to SCM**" Sanny adds all necessary info into the script body so when you decompile back you get the same source. Isn't it better than try to associate an external config file with a particular script?

**Comment by OrionSR** [comment]
No. In fact, I was going to pass on the recommendation that **Add extra info to SCM** be disabled and that **Write opcodes** be enabled by default.

The goal is to decompile the script, maintain the custom variables and reformat the script with different options or a different SCM.ini. The extra SCM info simply loads the crappy script I already have; nothing is decompiled. And I need the numbers if I want to debug a glitchy save. 

Then there's the complicated mess of TTDISA Vol 2, a main.scm modified as 3 different projects and has a giant gap in the variable space. They need to compile and decompile again to fix various issues, like decompiling with their main script without showing opcodes, but it is tough to tease out the custom variables in such a large script.

I don't have to perform this task often and I tend to work with standard SCMs, but when I do it's a difficult task. I'm a bit surprised I need to explain why a Make CustomVariable.ini is so critical for managing a custom main. Creating a custom edit mode seems like an obvious first step when working with a major mod like TTDISA, and CustomVariable.ini is a valuable component of the profile.
